### DISCUSSION TITLE
Migrate from monolithic docker to mongodb

### DISCUSSION THREAD

**Discussion by taker218** [OctoFarm/OctoFarm#1340] [General]
Hi everyone,

I'm currently running octofarm with the monolithic docker container and would like to migrate to the one with the seperate mongo db just to make things a bit more reliable.

What would be the best way to do that, so I don't loose the history and the settings I made.

Would it just be enough to create a new volume for the mongodb files and copy over the ones from the current octofarm-db volume?

**Comment by taker218** [comment]
Okay, I figured it out by tinkering a little bit :D

I'll try to write a step by step, how I did it:

1. Make backup from your mongodb in the monolithic-container:
- `docker exec -it octofarm-monolithic bash`
- cd into a directory you can reach from your docker host ( I used /app/scripts)
- `mongodump` (this makes a backup of your mongodb in the directory "dump" in the current folder)
- `exit`
2. Stop the octofarm container
3. move backup file from the one volume to one that you can reach from within the mongodb container
4. startup the new mongodb container and connect to it
- `docker exec -it mongodb bash`
- change into the directory where the backup is
- then the following command:
- `mongorestore --nsInclude=octofarm.* --username=MONGO_ROOTUSER_HERE --password=MONGO_PASSWORD_HERE dump/`
- let the backup finish
- exit the bash
5. startup the new octofarm container which uses the new mongodb and check if everything is there

**Reply by NotExpectedYet** [reply]
I was just writing this out for you! 

Thank you for the update on how to do it :) 

I will add this to the documentation page if you don't mind?

**Reply by NotExpectedYet** [reply]
The dump is the safer way to do it. 

I believe you should just be able to point your mongodb container at the octofarm/data directory though. 

Your way will cover you for mongodb major version upgrades and the lot though.

**Reply by taker218** [reply]
You can add this to the documentation page for sure. Might be handy for some people who just start with the monolithic one to check everything out and might want to migrate to a full mongodb one later :)

**Reply by taker218** [reply]
just added where to find the backup from the dump :)

**Reply by NotExpectedYet** [reply]
Brilliant thanks, I'll add it as some mongodb maintanence documation. 5 is out at the moment, and 6 soon and your method will work for updating to later versions with an extra command.
